src/MoBi.Core/OSPSuite.Core/Domain/Builder/CoreExpressionProfile.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using OSPSuite.Utility.Collections;
+using System.Collections.Generic;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using static OSPSuite.Core.CoreConstants.ContainerName;
+
+namespace OSPSuite.Core.Domain.Builder
+{
+   public class CoreExpressionProfile : BuildingBlock, IEnumerable<ExpressionParameter>
```

Comment:
I think this should implement the BuildingBlockStartValue T no? There is something like this for MoBi. Check ParameterStartValueBuildingBlock

Reply:
I think you meant BuildingBlock<T> which is what both StartValueBuildingBlocks do. If so, then you are right. Then I can get rid of all the enumerator stuff.

Reply:
no I just checked the code in core
it's called StartValueBuildingBlock<T>

Which ha all the behavior you need. While this building block base class can be renamed to, I think we should use this instead of reimplementing the mgt of list etc?